src/AspNet.Security.OAuth.WorkWeixin/AspNet.Security.OAuth.WorkWeixin/WorkWeixinAuthenticationHandler.cs

```diff
@@ -0,0 +1,152 @@
+using System.Security.Claims;
+using System.Text.Encodings.Web;
+using System.Text.Json;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.OAuth;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.WebUtilities;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace AspNet.Security.OAuth.WorkWeixin
+{
+    public class WorkWeixinAuthenticationHandler : OAuthHandler<WorkWeixinAuthenticationOptions>
+    {
+        public WorkWeixinAuthenticationHandler(
+        [NotNull] IOptionsMonitor<WorkWeixinAuthenticationOptions> options,
+        [NotNull] ILoggerFactory logger,
+        [NotNull] UrlEncoder encoder,
+        [NotNull] ISystemClock clock)
+        : base(options, logger, encoder, clock)
+        {
+        }
+
+        protected override async Task<AuthenticationTicket> CreateTicketAsync(
+            [NotNull] ClaimsIdentity identity,
+            [NotNull] AuthenticationProperties properties,
+            [NotNull] OAuthTokenResponse tokens)
+        {
+            (int errCode, string? userId) = await GetUserIdentityfierAsync(tokens);
+            if (errCode != 0 || string.IsNullOrEmpty(userId))
+            {
+                throw new HttpRequestException($"An error (Code:{errCode}) occurred while retrieving the user identitfier");
+            }
+
+            string address = QueryHelpers.AddQueryString(Options.UserInformationEndpoint, new Dictionary<string, string?>
+            {
+                ["access_token"] = tokens.AccessToken,
+                ["userid"] = userId,
+            });
+
+            using var response = await Backchannel.GetAsync(address);
+            if (!response.IsSuccessStatusCode)
+            {
+                Logger.LogError("An error occurred while retrieving the user profile: the remote server " +
+                                "returned a {Status} response with the following payload: {Headers} {Body}.",
+                                /* Status: */ response.StatusCode,
+                                /* Headers: */ response.Headers.ToString(),
+                                /* Body: */ await response.Content.ReadAsStringAsync(Context.RequestAborted));
+
+                throw new HttpRequestException("An error occurred while retrieving user information.");
+            }
+
+            using var payload = JsonDocument.Parse(await response.Content.ReadAsStringAsync(Context.RequestAborted));
+
+            errCode = payload.RootElement.GetProperty("errcode").GetInt32();
+            if (errCode != 0)
+            {
+                Logger.LogError("An error occurred while retrieving the user profile: the remote server " +
+                          "returned a {Status} response with the following message: {Message} .",
+                          /* Status: */ errCode,
+                          /* Message: */ payload.RootElement.GetString("errmsg"));
```

Comment:
The error didn't occur retrieving it - it was that there was an error code in the response.